src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/Eventing/AsyncParameterCapturingEventSource.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.StartupHook;
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.Eventing
+{
+    internal sealed class AsyncParameterCapturingEventSource : IDisposable
+    {
+        private readonly Thread _thread;
+        private readonly CancellationTokenSource _cts = new();
+        private readonly BlockingCollection<Action<CancellationToken>> _pendingEvents;
+        private readonly ParameterCapturingEventSource _eventSource;
+
+        private const int MaxPendingEventsCollectionCapacity = 1024;
+        private const string BackgroundLoggingThreadName = "[dotnet-monitor] Probe EventSource Thread";
+
+        public AsyncParameterCapturingEventSource(ParameterCapturingEventSource eventSource)
+        {
+            _pendingEvents = new BlockingCollection<Action<CancellationToken>>(MaxPendingEventsCollectionCapacity);
```

Comment:
nit: Let's initialize this inline with its declaration just like the `CancellationTokenSource`